salalib/shapemap.cpp

```diff
@@ -1577,7 +1575,7 @@ void ShapeMap::polyInPolyList(int polyref, pvecint& shapeindexlist, double toler
                for (const ShapeRef& shapeRef: shaperefs) {
                   if (*iter != shapeRef && ((iter->m_tags & ShapeRef::SHAPE_CENTRE) || (shapeRef.m_tags & ShapeRef::SHAPE_CENTRE))) {
                      if (testedlist.add(shapeRef.m_shape_ref) != -1) {
-                        shapeindexlist.add(depthmapX::findIndexFromKey(m_shapes, (int) shapeRef.m_shape_ref));
+                        depthmapX::insert_sorted(shapeindexlist, int(depthmapX::findIndexFromKey(m_shapes, int(shapeRef.m_shape_ref))));

```

Comment:
The fact that we do a linear search across a map to then insert the index via a binary search into a sorted index vector is a bit of a code smell - but fine for this PR... Maybe raise a ticket for this?